eventuals/task.h

```diff
@@ -238,34 +254,56 @@ struct _TaskFromToWith {
     K_ k_;
     std::tuple<Args_...> args_;
 
-    Callback<
-        Action,
-        std::optional<std::exception_ptr>&&,
-        Args_&&...,
-        std::optional<
-            std::conditional_t<
-                std::is_void_v<From_>,
-                std::monostate,
-                From_>>&&,
-        std::unique_ptr<void, Callback<void*>>&,
-        Interrupt&,
+    // The 'dispatch_' is a `std::variant` because its either a function that
+    // creates an eventual or the value from 'Task::Success' that should be
+    // passed on to the continuation.
+    std::variant<
         std::conditional_t<
             std::is_void_v<To_>,
-            Callback<>&&,
-            Callback<To_>&&>,
-        Callback<std::exception_ptr>&&,
-        Callback<>&&>
+            std::monostate,
+            // Using std::decay_t because std::variant doesn't allow storing
```

Comment:
How will this work for non-copyable types (e.g. `std::unique_ptr`) and types which are not moveable or copyable (e.g. [`std::mutex`](https://en.cppreference.com/w/cpp/thread/mutex))? Maybe expand this comment.

(I wonder if [std::reference_wrapper](https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper) would work if we want to store a reference)

Reply:
Now this will be applicable only for moveable types. Will add std::reference_wrapper in the future if needed.